tests/mu_define_enum_test.c

```diff
@@ -34,5 +66,62 @@ int run_mu_define_enum_tests(void)
 
     POOR_MANS_ASSERT(MU_ENUM_VALUE_COUNT(TEST4_ENUM_VALUES) == 3);
 
+    const char* TEST5_test5_b = MU_ENUM_TO_STRING(TEST5_ENUM, 42);
+    POOR_MANS_ASSERT(TEST5_test5_b != NULL);
+    POOR_MANS_ASSERT(strcmp("test5_b = 42", TEST5_test5_b) == 0);
+
+    const char* TEST5_unknown = MU_ENUM_TO_STRING(TEST5_ENUM, 41);
+    POOR_MANS_ASSERT(TEST5_unknown != NULL);
+    POOR_MANS_ASSERT(strcmp("UNKNOWN", TEST5_unknown) == 0);
+
+    const char* TEST6_test6_a = MU_ENUM_TO_STRING(TEST6_ENUM, test6_a);
+    POOR_MANS_ASSERT(TEST6_test6_a != NULL);
+    POOR_MANS_ASSERT(strcmp("test6_a = 42", TEST6_test6_a) == 0);
+
+    // test6_b and TEST6_ENUM_INVALID are both 0
+    const char* TEST6_test6_b = MU_ENUM_TO_STRING(TEST6_ENUM, test6_b);
+    POOR_MANS_ASSERT(TEST6_test6_b != NULL);
+    POOR_MANS_ASSERT(strcmp("TEST6_ENUM_INVALID", TEST6_test6_b) == 0);
```

Comment:
>TEST6_ENUM_INVALID [](http://example.com/codeflow?start=29&length=18)

Why is it expected that the most unclear value (we sneak in the _INVALID) would be printed here? I would say _INVALID is OK to be printed if and only if none of the values in the enum are "0" naturally somehow.

Reply:
same comment for below... for test6_d

Reply:
Well in any enum that has explictly a value of 0 we will NOT be able to discriminate between INVALID (which is 0) and an explicit 0.

---
In reply to: [632130120](https://github.com/Azure/macro-utils-c/pull/30#discussion_r632130120) [](ancestors = 632130120)